Dev4Agriculture.ISO11783.ISOXML/Geometry/GeometryUtility.cs

```diff
@@ -0,0 +1,222 @@
+                return Math.Abs(pt1.PointEast - pt.PointEast) < epsilon || Math.Abs(pt2.PointEast - pt.PointEast) < epsilon;
+
+            var x = pt1.PointNorth + (pt.PointEast - pt1.PointEast) * (pt2.PointNorth - pt1.PointNorth) / (pt2.PointEast - pt1.PointEast);
+            var y = pt1.PointEast + (pt.PointNorth - pt1.PointNorth) * (pt2.PointEast - pt1.PointEast) / (pt2.PointNorth - pt1.PointNorth);
+
+            return Math.Abs(pt.PointNorth - x) < epsilon || Math.Abs(pt.PointEast - y) < epsilon;
+        }
+
+        public static bool IsPolygonsEqual(List<ISOPoint> polygon1, List<ISOPoint> polygon2)
+        {
+            if (polygon1.Count != polygon2.Count)
+            {
+                return false;
+            }
+            for (var i = 0; i < polygon1.Count; i++)
+            {
+                var p1 = polygon1[i];
+                var p2 = polygon2[i];
+                if (p1.PointNorth != p2.PointNorth || p1.PointEast != p2.PointEast)
+                {
+                    return false;
+                }
+            }
+            return true;
+        }
+
+        private static List<ISOPoint> OrderPointsByDistanceFromCenter(this List<ISOPoint> points)
+        {
+            if (!points.Any())
+                return points;
+            decimal mX = 0;
+            decimal my = 0;
+            foreach (var p in points)
+            {
+                mX += p.PointEast;
+                my += p.PointNorth;
+            }
+            mX /= points.Count();
+            my /= points.Count();
+            return points.OrderBy(v => Math.Atan2((double)(v.PointNorth - my), (double)(v.PointEast - mX))).ToList();
+        }
+
+        public static List<ISOPoint> GetIntersectionOfPolygons(this List<ISOPoint> poly1, List<ISOPoint> poly2)
+        {
+            var clippedCorners = new List<ISOPoint>{};
+            //Add  the corners of poly1 which are inside poly2       
+            foreach (var t in poly1.Where(t => IsPointInsidePoly(t, poly2)))
+            {
+                AddPoints(clippedCorners, new[] { t });
+            }
+
+            //Add the corners of poly2 which are inside poly1
+            foreach (var t in poly2.Where(t => IsPointInsidePoly(t, poly1)))
+            {
+                AddPoints(clippedCorners, new[] { t });
+            }
+
+            //Add  the intersection points
+            for (int i = 0, next = 1; i < poly1.Count(); i++, next = (i + 1 == poly1.Count()) ? 0 : i + 1)
+            {
+                AddPoints(clippedCorners, GetIntersectionPoints(poly1[i], poly1[next], poly2));
+            }
+
+            var res = new List<ISOPoint>();
+            res.AddRange(clippedCorners.OrderPointsByDistanceFromCenter());
+            return res;
+        }
+
+        private static bool IsEqual(decimal d1, decimal d2) => Math.Abs(d1 - d2) <= EquityTolerance;
+
+        private static bool IsPointInsidePoly(ISOPoint test, List<ISOPoint> poly)
```

Comment:
We have a function IsInPolygon in ISOPolygon (see ISOExtensions), can we reuse this here?